File: src/ForegroundModule.ts
import { NativeModule, requireNativeModule } from "expo";

import { ForegroundModuleEvents } from "./Foreground.types";

declare class ForegroundModule extends NativeModule<ForegroundModuleEvents> {
  PI: number;
  hello(): string;
  setValueAsync(value: string): Promise<void>;
  startForegroundService: (
    endpoint: string,
    title: string,
    subtext: string
  ) => Promise<void>;
  stopForegroundService: () => Promise<void>;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<ForegroundModule>("Foreground");

File: ios/ForegroundModule.swift
import ExpoModulesCore
import UserNotifications

public class ForegroundModule: Module {
  public func definition() -> ModuleDefinition {
    Name("Foreground")

    Function("startForegroundService") { (endpoint: String, title: String, subtext: String, progress: Int) -> String in
      return "start"
    }

    Function("stopForegroundService") { () -> String in
      return "stop"
    }
  }
}
